pkg/storage/unified/resource/server.go

```diff
@@ -450,8 +455,24 @@ func (s *server) newEvent(ctx context.Context, user claims.AuthInfo, key *resour
 		return nil, AsErrorResult(err)
 	}
 	if len(secure) > 0 {
-		// See: https://github.com/grafana/grafana/pull/107803
-		return nil, NewBadRequestError("Saving secure values is not yet supported")
+		// Make sure the secure values are safe to save (just in case)
+		for _, v := range secure {
+			if !v.Create.IsZero() {
+				return nil, NewBadRequestError("unable to create values in unified storage")
+			}
+			if v.Remove {
+				return nil, NewBadRequestError("unable to save remove command")
+			}
+			if v.Name == "" {
+				return nil, NewBadRequestError("secure value requires name")
+			}
+		}
+		if s.secure == nil {
+			return nil, AsErrorResult(fmt.Errorf("secure value store not configured"))
+		}
+		if err := s.secure.CanReference(ctx, utils.ToObjectReference(obj), secure); err != nil {
```

Comment:
I think we should only make this check if the the references have changed; If the oldValue secrets are different from the newValue secrets. Otherwise a user making a change to another field (i.e adding an annotation) without touching the reference will be denied the object unless the user can also reference the existing secret.